src/CompoundRegion.cc

```diff
@@ -61,28 +61,38 @@ std::uint64_t consumeDecodeU64(std::uint8_t const *&buffer, std::uint8_t const *
 template <typename F>
 auto getUnionBounds(UnionRegion const &compound, F func) {
     auto bounds = func(compound.getOperand(0));
-    bounds.expandTo(func(compound.getOperand(1)));
+    for (unsigned i = 1; i < compound.nOperands(); ++ i) {
+        bounds.expandTo(func(compound.getOperand(i)));
+    }
     return bounds;
 }
 
 template <typename F>
 auto getIntersectionBounds(IntersectionRegion const &compound, F func) {
     auto bounds = func(compound.getOperand(0));
-    bounds.clipTo(func(compound.getOperand(1)));
+    for (unsigned i = 1; i < compound.nOperands(); ++ i) {
+        bounds.clipTo(func(compound.getOperand(i)));
+    }
     return bounds;
 }
 
 }  // namespace
 
-CompoundRegion::CompoundRegion(Region const &first, Region const &second)
-        : _operands{first.clone(), second.clone()} {}
-
-CompoundRegion::CompoundRegion(
-    std::array<std::unique_ptr<Region>, 2> operands) noexcept
-        : _operands(std::move(operands)) {}
+CompoundRegion::CompoundRegion(std::vector<std::unique_ptr<Region>> operands) noexcept
+        : _operands(std::move(operands))
+{
```

Comment:
It might be worth it to have some logic here (or maybe in subclass constructors) to look for operands that are themselves compound regions of the same type, in order to flatten them out.

Reply:
That logic would also need to handle nested unions and intersections differently, I guess?

Reply:
Yes, but only in the sense that we don't try to simplify a compound region of the opposite type.  I don't think we need to go looking for intersections buried under unions inside an intersection, for instance; if we did I think it'd be very easy to do more harm than good.